### ISSUE TITLE
AI spawn bug

### ISSUE THREAD

**Issue by xarsak** [goonstation/goonstation#7828]
### Labels

[BUG]

### Description

You dont get respawned as AI after being spawned as an ai twice if you have puritan

### Steps to reproduce

1. Die
2.have someone take your brain 
3.have this someone insert that brain in an ai core
4.have this someone remove your brain from ai core after activating you
5.take that brain back to your body
6.clone your body
7.die from puritan
8.take your original bodys brain back to ai cell
9.you dont get revived but the ai lights up normally like you did

### Expected Behavior

I expected to be revived as an ai

### Additional Information & Screenshots

i wasnt scanned before i got cloned and we talked a bit before they removed the brain

**Comment by Zamazenta** [comment]
I believe this is because once cloned, your old brain will no longer be valid. In order to be revived as an AI after being cloned they would need the brain from the cloned body, not the original body.

**Comment by frawhst** [comment]
Yeah, this seems normal. Once you're cloned, your 'current' brain is in the cloned body, and the old brain won't work. If you're PURITAN cloned, then your new brain.... explodes, so there's no brain left to borg you with.
Will leave open since this can be unintuitive and might be worth changing.

**Comment by TheTyrant25** [comment]
the naming of this issue is very unclear, ideally have a dev rename it to something like "cloned puritan to ai respawn failure".

Also if the brain is a cold one then it shouldn't have seemingly activated the ai should it.
